oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/server/WebSessionOAuth2ServerAuthorizationRequestRepository.java

```diff
@@ -53,7 +53,7 @@
 		if (state == null) {
 			return Mono.empty();
 		}
-		return getStateToAuthorizationRequest(exchange, false)
```

Comment:
This looks like it breaks functionality. It should not add an empty Map if we are loading a value and nothing exists.

Reply:
The behaviour here is kept the same as before and I only split `getStateToAuthorizationRequest(ServerWebExchange exchange, boolean create)` into `getStateToAuthorizationRequest(ServerWebExchange exchange)` and `saveStateToAuthorizationRequest(ServerWebExchange exchange)` to make it clear.

Reply:
And the fix is in `saveStateToAuthorizationRequest`